components/ai_chat/core/engine/remote_completion_client.cc

```diff
@@ -139,7 +126,30 @@ void RemoteCompletionClient::QueryPrompt(
     EngineConsumer::GenerationCompletedCallback data_completed_callback,
     EngineConsumer::GenerationDataCallback
         data_received_callback /* = base::NullCallback() */) {
-  const GURL api_base_url = GetEndpointBaseUrl();
+  auto callback = base::BindOnce(
+      &RemoteCompletionClient::OnFetchPremiumCredential,
+      weak_ptr_factory_.GetWeakPtr(), prompt, extra_stop_sequences,
+      std::move(data_completed_callback), std::move(data_received_callback));
+  credential_manager_->FetchPremiumCredential(std::move(callback));
+}
+
+void RemoteCompletionClient::OnFetchPremiumCredential(
+    const std::string& prompt,
+    const std::vector<std::string> extra_stop_sequences,
+    EngineConsumer::GenerationCompletedCallback data_completed_callback,
+    EngineConsumer::GenerationDataCallback data_received_callback,
+    absl::optional<CredentialCacheEntry> credential) {
+  bool premium_enabled = credential.has_value();
+  const GURL api_base_url = GetEndpointBaseUrl(premium_enabled);
+  base::flat_map<std::string, std::string> headers;
+  if (premium_enabled) {
+    // Add Leo premium SKU credential as a Cookie header.
+    std::string cookie_header_value =
+        "__Secure-sku#brave-leo-premium=" + credential->credential;
```

Comment:
Isn't this a JSON API?
Shouldn't we rather use a `Authorization:` header or a specific `X-Brave-Premium:` header?

Reply:
I talked with @evq about this, we decided to send as a cookie rather than an Authorization header for consistency.  Currently the subscription service [accepts](https://github.com/brave-intl/subscriptions/blob/ad811bf4c8f3063bf4c0012f2d29f6f2f7b2d7b6/pkg/api/middleware.go#L132-L206) the credential as a cookie from the Client (which I believe is Guardian). So we're just doing the same here.